canvas_oauth/tests/test_settings.py

```diff
@@ -0,0 +1,52 @@
+from django.test import TestCase
+from django.conf import settings
+from django.core.exceptions import ImproperlyConfigured
+from contextlib import contextmanager
+import importlib
+import datetime
+
+
+CANVAS_OAUTH_SETTINGS_MODULE_NAME = "canvas_oauth.settings"
+
+
+@contextmanager
+def required_oauth_settings(oauth_settings={}):
+    settings.CANVAS_OAUTH_CLIENT_ID = oauth_settings.get('CANVAS_OAUTH_CLIENT_ID', '10000000000012')
+    settings.CANVAS_OAUTH_CLIENT_SECRET = oauth_settings.get('CANVAS_OAUTH_CLIENT_SECRET', 'yKZc1bJpdykVBUT4')
+    settings.CANVAS_OAUTH_CANVAS_DOMAIN = oauth_settings.get('CANVAS_OAUTH_CANVAS_DOMAIN', 'canvas.localhost')
+    yield
```

Comment:
Wow. Cool. TBH I didn't really think you could just add and remove members of `settings` without somehow messing up Django. GTK. (In `pytest` you can do some of this using fixtures.)

Reply:
GTK that this might be more easily accomplished in `pytest` with some basic fixtures.